yarn-project/aztec-node/src/sentinel/sentinel.ts

```diff
@@ -375,6 +376,35 @@ export class Sentinel extends (EventEmitter as new () => WatcherEmitter) impleme
     };
   }
 
+  /** Computes stats for a single validator. */
+  public async getValidatorStats(
+    validatorAddress: string,
+    fromSlot?: bigint,
+    toSlot?: bigint,
+  ): Promise<SingleValidatorStats | undefined> {
+    const ethAddress = EthAddress.fromString(validatorAddress);
+    const history = await this.store.getHistory(ethAddress);
+
+    if (!history || history.length === 0) {
+      return undefined;
+    }
+
+    const slotNow = this.epochCache.getEpochAndSlotNow().slot;
+    const effectiveFromSlot = fromSlot ?? (this.lastProcessedSlot ?? slotNow) - BigInt(this.store.getHistoryLength());
+    const effectiveToSlot = toSlot ?? this.lastProcessedSlot ?? slotNow;
+
+    const validator = this.computeStatsForValidator(validatorAddress as `0x${string}`, history, effectiveFromSlot, effectiveToSlot);
+    const provenPerformance = await this.store.getProvenPerformance(ethAddress);
```

Comment:
Note that proven performance is returned for the entire history, and not filtered by the slot range requested. Is this intentional? If so, let's rename to something that makes it clear, like `completeProvenPerformance` or `allProvenPerformance` (I don't really like those, so feel free to change if you come up with a better one!).